lib/logstash/inputs/base.rb

```diff
@@ -100,6 +101,20 @@ def tag(newtag)
     @tags << newtag
   end # def tag
 
+  # if you override stop, don't forget to call super
+  # as the first action
+  public
+  def stop
+    @logger.debug("stopping", :plugin => self)
```

Comment:
suggestion: maybe log "stop called" instead of "stopping" it seems to better represent the reality?